lib/croma/struct.ex

```diff
@@ -133,6 +158,32 @@ defmodule Croma.Struct do
     end
   end
 
+  defp evaluate_existing_field(mod, value, false), do: mod.validate(value)
+  defp evaluate_existing_field(mod, value, true )  do
+    R.or_else(mod.validate(value), try do
+      mod.new(value)
+    rescue
+      _ -> {:error, {:invalid_value, [mod]}}
+    end)
+  end
+
+  defp evaluate_non_existing_field(mod, false), do: try_default(mod)
+  defp evaluate_non_existing_field(mod, true )  do
+    R.or_else(try_default(mod), try do
+      mod.new(%{})
+    rescue
+      _ -> {:error, {:value_missing, [mod]}}
+    end)
```

Comment:
Let's define `try_new(mod, value)` to put the common part of `evaluate_existing_field/2` and `evaluate_non_existing_field/2` in one place.